### DISCUSSION TITLE
posts/aws-api-gateway-cheatsheet

### DISCUSSION THREAD

**Discussion by giscus** [ncoughlin/ncoughlin-giscus#2] [Announcements]
# posts/aws-api-gateway-cheatsheet

A reference for common Integration Request mapping template variables

https://ncoughlin.com/posts/aws-api-gateway-cheatsheet

**Comment by adminy** [comment]
Does 
```json
{
  "body" : $input.json('$') 
}
```
Work for missing body and how to do for non application/json type templates? e.g. image/png or application/xml?

**Reply by ncoughlin** [reply]
I believe that this would evaluate to null, which can cause errors downstream if not accounted for. You can set a default value though:

```
#set($body = $input.body)
{
  "body" : $input.json('$') ?: "default value"
}
```

I don't know about image/png or xml though. If you find out let me know and I can add it to the article.
